Update supported version documentation (#1860)
pulumi/pulumi-hugo: themes/default/content/docs/intro/languages/java.md

<img src="/logos/tech/java.svg" align="right" width="90" style="padding:8px; margin-top: -64px">

> Pulumi supports writing your infrastructure as code using the Java language running on any [Supported version](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) after Java 11.

Reviewer: ```suggestion
Pulumi supports writing your infrastructure as code using the Java language running on any [supported version](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) after Java 11.
```

Reviewer 2: "After 11" reads a bit ambiguously to me. (My first thought was, _Oh, I thought we supported 11 also. Do we only support 12 and up?_) If we do mean "11 or later", we should probably say that.

Author: It's 11 or later. So "Pulumi supports writing your infrastructure as code using the Java language running on any supported version after Java 11 or later." ?

> Pulumi supports writing your infrastructure as code using the Java language running on any [supported version](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) after Java 11.

Author: ```suggestion
Pulumi supports writing your infrastructure as code using the Java language running on any [supported version](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) after Java 11 or later.
```

Author: Does this read ok?

> Pulumi supports writing your infrastructure as code using the Java language running on any [supported version](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) after Java 11.

Reviewer 2: `after` was the problematic word, so I might just suggest `of` instead:

```suggestion
Pulumi supports writing your infrastructure as code using the Java language running on any [supported version](https://www.oracle.com/java/technologies/java-se-support-roadmap.html) of Java 11 or later.
```

What do you think?

> We support running Maven Pulumi programs, Gradle Pulumi programs, and Pulumi programs packaged as JAR files.

Reviewer 3: Great start. FWIW Maven and Gradle have documented lower bounds https://github.com/pulumi/pulumi-java#requirements  - but I don't think we figured out the LTS strategy yet. We can refine this later. LGTM.
